### DISCUSSION TITLE
query error typing help

### DISCUSSION THREAD

**Discussion by zhihengGet** [reduxjs/redux-toolkit#1579] [General]
hello, thanks for the help

i use custom `queryFn `and i typed the query result as

```
Promise<
  | {
      data: {};
      error?: undefined;
    }
  | {
      data?: undefined;
      error: { text: string; statusCode: number };
      meta?: { request: Request; response: Response };
    }
>
```
 
the thing is that i when i access` result.error.text` i keep getting `SerializedError `does not have text 

how i remove this `SerializedError `from ts

**Comment by phryneas** [comment]
There is always the possibility of a `SerializedError` since we cannot "read" your code and guarantee that it doesn't just `throw` an exception at some point - and every unhandled error would be handled as a `SerializedError`.

It's kinda how in TypeScript, in a `catch` block, the error is either `any` or `unknown`, but never something specific until you start making assumptions on it.

So you will probably need to write some [Type Guard function](https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards) to make sure it is what you expect.

**Reply by bggolding** [reply]
This is a really common question that comes up for me a lot. While I know how to write a type guard, I never know what the correct test to use for the type guard is, which types I should be checking against in RTK or whether there’s pre-existing support in the library for this. Could we perhaps get an example?

I find handling and reporting errors from queries to be really difficult. I wind up in a mess of tests whether there are attributes on the error object and hope that I’ve managed to get the right ones together so that the error appears coherent. When I contrast this with handling errors from a mutation using (say) `.catch((error) => setErrorMessage(error.data.message))` I just reckon this could be better.
